app/parse.py

```diff
@@ -15,8 +41,63 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def parse_single_product(product: WebElement) -> Product:
+    return Product(
+        title=(
+            product.find_element(By.CLASS_NAME, "title").get_attribute("title")
+        ),
+        description=product.find_element(By.CLASS_NAME, "description").text,
+        price=float(
+            product.find_element(By.CLASS_NAME, "price")
+            .text.replace("$", "")
+        ),
+        rating=len(product.find_elements(By.CLASS_NAME, "glyphicon-star")),
+        num_of_reviews=int(
+            product.find_element(By.CLASS_NAME, "ratings")
+            .find_element(By.TAG_NAME, "p")
+            .text.split()[0]
+        )
+    )
+
+
+def get_all_url_products(page_url: str, driver_instance: Driver) -> list[WebElement]:
+    driver_instance.driver.get(page_url)
+    sleep(2)
+
+    try:
+        driver_instance.driver.find_element(By.ID, "closeCookieBanner").click()
+    except NoSuchElementException:
+        pass
```

Comment:
it's probably better to print something like: Could not find element in page_url